src/demo/demo.js

```diff
@@ -101,3 +55,80 @@ const getToken = function(onSuccess) {
   }
   request.send()
 }
+
+
+class SDK extends Component{
+  componentDidMount () {
+    this.initSDK(this.props.options)
+  }
+
+  componentWillReceiveProps({options}){
+    if (this.state.onfidoSdk){
+      this.state.onfidoSdk.setOptions(options)
+    }
+  }
+
+  componentWillUnmount () {
+    if (this.state.onfidoSdk) this.state.onfidoSdk.tearDown()
+  }
+
+  initSDK = (options)=> {
+    const onfidoSdk = Onfido.init(options)
+    this.setState({onfidoSdk})
```

Comment:
what's your opinion on using react state for this, rather than just a class property?
I generally only use React state when the thing I'm storing could possibly need to trigger a re-render when being set (which isn't the case here)

Reply:
to be honest, I don't tend to get to that level of optimisation, I tend to follow what reduces the changes of unsuspecting bugs and if there is indeed a performance issue then I optimise. In this case I think `onfidoSDK` will never have to trigger a render, because I even have `shouldComponentUpdate -> false`, so it really doesn't have to be in state. However, this also means that having it on state won't trigger a render either.

Reply:
I don't do it for purely optimisation reasons
In my mind it's an indicator to the reader if this particular variable affects the render or not

Reply:
No, it's normally fairly easy to spot - the only time the assumption changes is when the variable enters the `render` method

At which point you're writing `this.myVar` in the render method, and you think "Oh, that needs to be stateful"

But yeah, it's a fair point - I'm not sure if more junior developers would also think that. So maybe it's not worth it, for that reason 🤔